### PR TITLE
[#150832726] - implement offer to invite guest user as full account w…

### PR DESCRIPTION
**Pull request by kravchina**
…hen changes are made to referral by destination users

### CODE REVIEW

File: src/js/controllers/viewReferralsController.js [hunk 1]
```diff
@@ -315,7 +321,24 @@ angular.module('viewReferrals')
             modalInstance.result.then(function() {
                 $scope.referral.attachments.splice($scope.referral.attachments.indexOf(attachment),1);
                 Attachment.delete({id: attachment.id});
+                $scope.showOfferToInviteGuest();
             });
         };
+
+        $scope.showOfferToInviteGuest = function () {
+            if($scope.referral.requires_guest_activation) {
```
Comment (anton-kaidalov) [inline src/js/controllers/viewReferralsController.js]: This `if` effectively blocks the usage of this feature :) If referral requires activation, it means it shouldn't be shown anywhere, so how could anybody edit it?

What we should check here is whether the original provider is guest or not.

File: src/js/services/services.js [hunk 2]
```diff
@@ -137,7 +137,8 @@ angular.module('dentalLinksServices')
         findByPractice: {method: 'GET', url: API_ENDPOINT + '/referrals/practice/:id', isArray: false},
         countByInvited: {method: 'GET', url: API_ENDPOINT + '/referrals/count/:id', isArray: false},
         countByPractice: {method: 'GET', url: API_ENDPOINT + '/referrals/count_by_practice/:id', isArray: false},
-        createGuestReferral: {method: 'POST', url: API_ENDPOINT + '/referrals/new/guest_referral', isArray: false}
+        createGuestReferral: {method: 'POST', url: API_ENDPOINT + '/referrals/new/guest_referral', isArray: false},
+        inviteGuest: {method: 'GET', url: API_ENDPOINT + '/referrals/:id/invite_guest'}
```
Comment (anton-kaidalov) [inline src/js/services/services.js]: Don't forget to rename this according to the backend endpoint.

File: src/js/controllers/viewReferralsController.js [hunk 3]
```diff
@@ -315,7 +321,24 @@ angular.module('viewReferrals')
             modalInstance.result.then(function() {
                 $scope.referral.attachments.splice($scope.referral.attachments.indexOf(attachment),1);
                 Attachment.delete({id: attachment.id});
+                $scope.showOfferToInviteGuest();
             });
         };
+
+        $scope.showOfferToInviteGuest = function () {
```
Comment (anton-kaidalov) [inline src/js/controllers/viewReferralsController.js]: You added this function for the View Referral page, but notes are also used on Create Guest Referral. This causes Create Guest Referral to crash on adding a note.

Please search and check where else the notes block is used. Create Referral, for example? It probably crashes as well.

File: src/js/controllers/notesController.js [hunk 4]
```diff
@@ -21,6 +21,7 @@ angular.module('dentalLinks')
             });
             ModalHandler.set(modalInstance);
             modalInstance.result.then(function (message) {
+                $scope.showOfferToInviteGuest();
```
Comment (anton-kaidalov) [inline src/js/controllers/notesController.js]: Does it make sense to do this here, before we even know if it's the `immediateUpdate`? Are there situations when it's not, but we still need to show the offer?

File: src/js/controllers/notesController.js [hunk 5]
```diff
@@ -78,6 +79,7 @@ angular.module('dentalLinks')
                 }
             });
             modalInstance.result.then(function (note) {
+                $scope.showOfferToInviteGuest();
```
Comment (anton-kaidalov) [inline src/js/controllers/notesController.js]: Same comment as here: https://github.com/avidmich/referral-frontend/pull/261/files#diff-e4283bdd12368b2cf3fcc32f4c675ba1R24

File: src/js/controllers/viewReferralsController.js [hunk 6]
```diff
@@ -239,6 +241,7 @@ angular.module('viewReferrals')
             ModalHandler.set(modalInstance);
             modalInstance.result.then(function (patient) {
                 $scope.referral.patient = patient;
+                $scope.showOfferToInviteGuest();
```
Comment (anton-kaidalov) [inline src/js/controllers/viewReferralsController.js]: How is it possible to trigger this offer?

File: src/js/controllers/viewReferralsController.js [hunk 7]
```diff
@@ -256,6 +259,7 @@ angular.module('viewReferrals')
             ModalHandler.set(modalInstance);
             modalInstance.result.then(function (providerId) {
                 $scope.initModel();  // have to reload referrer object from server, because it needs to provide object that has dest_provider attribute and many other customizations.
+                $scope.showOfferToInviteGuest();
```
Comment (anton-kaidalov) [inline src/js/controllers/viewReferralsController.js]: Doesn't show the dialog for me.

File: src/js/controllers/viewReferralsController.js (hunk 6 above)
Comment (kravchina) [reply src/js/controllers/viewReferralsController.js]: Removed.

File: src/js/controllers/viewReferralsController.js (hunk 7 above)
Comment (kravchina) [reply src/js/controllers/viewReferralsController.js]: Fixed.

File: src/js/controllers/viewReferralsController.js (hunk 1 above)
Comment (kravchina) [reply src/js/controllers/viewReferralsController.js]: Fixed.

File: src/js/services/services.js (hunk 2 above)
Comment (kravchina) [reply src/js/services/services.js]: Fixed.

File: src/js/controllers/viewReferralsController.js (hunk 3 above)
Comment (kravchina) [reply src/js/controllers/viewReferralsController.js]: Fixed

**Comment by anton-kaidalov** [comment]
Slava, thank you. No more changes needed from your side.

We will proceed forward the following way.

**Don't merge this into master.**

I'll merge into my guest-conversion branch, integrate there, and then release the result to Alexei.

**Comment by anton-kaidalov** [comment]
For now I'll close this PR, but please don't delete the branches right now. I'll merge and then let you know when to delete.

**Pull request closed without merging**